lua/harpoon/init.lua

```diff
@@ -34,6 +34,11 @@ HarpoonConfig = HarpoonConfig or {}
 -- tbl_deep_extend does not work the way you would think
 local function merge_table_impl(t1, t2)
     for k, v in pairs(t2) do
+        if (k == 'base_dirs') then
```

Comment:
- if I add an or condition to include "menu"
- and assuming that menu = {} in the harpoon setup call
then #47 can also be fixed ... fyi @jesseleite

Reply:
Heyo! So if I delete my custom `menu = { ... }` config (to revert to harpoon defaults in my setup method), would I still have to go and bust the cache myself somewhere? Hoping not 🙏

Reply:
yups,
- originally, if you did this to set up your menu
``` lua
require("harpoon").setup({
    menu = {
        width = vim.api.nvim_win_get_width(0) - 4,
    }
-- other settings ... 
})
```
- later to remove your menu settings, you can do this
``` lua
-- need to call setup again and pass an empty table to the menu key to clear the menu settings
require("harpoon").setup({
    menu = {}
-- other settings ... 
})
```
  - unfortunately this won't work (removing the menu key entirely)
``` lua
-- need to call setup again and pass an empty table to the menu key to clear the menu settings
require("harpoon").setup({
-- other settings ... 
})
```
no need to mess about with deleting the harpoon cache

---

From my preliminary testing, providing menu={} clears up the menu settings without needing to mess about with the cache.

This is caused by the current deep merge implementation which doesn't merge nicely. The nicer solution would be to clean up the deep merge function. The work around is to detect if we are attempting to merge the menu key and simply overwrite the 'left' table. Haven't created a PR for this yet though ... but should be simple enough to add an or condition and PR ...

Reply:
Right, though I disagree with having to leave `menu = {}` to clear some cache. What I'm suggesting in #47 is to altogether avoid caching configs that don't benefit from caching in the first place. Should people really have to provide an empty object in order to revert back to `setup` function defaults? 🙂

Reply:
yeah, agree that it's not ideal. Seems like the deep merge function would also be a good excuse to have a go at implementing unit tests ... #30 ... will get round to it if I get round to it :)

Reply:
![](https://chooice.co.nz/wp-content/uploads/2020/10/Rimu-Round-Tuit-scaled.jpg)